### ISSUE TITLE
Feature Request: Support SSR

### ISSUE THREAD

**Issue by matthew-dean** [vuejs/rollup-plugin-vue#101]
Rollup-plugin-vue relies on the `vue-template-compiler` vs `vue-server-renderer` package. As far as I can tell, one major distinction is that `vue-server-render` sets this line:
```
process.env.VUE_ENV = 'server';
```
One of the effects is that it allows you to output things like `<script>` tags from your template (rather than being stripped). I use using Rollup to generate static HTML files (that get hydrated on load in the browser to a "live" Vue component) based on https://ssr.vuejs.org/en.

So, basically, a feature request would be able to pass something like `server: true` to rollup-plugin-vue, and have it set `process.env.VUE_ENV = 'server'`. I know it's a small thing, but hopefully a documented feature would save someone else trying to figure out why Rollup can't generate static HTML using this plugin.

**Comment by znck** [comment]
@matthew-dean @Justineo Could you explain this requirement a bit more? What are the use cases and how to reproduce the issue you are facing?

**Comment by Justineo** [comment]
@znck if a project depends on certain CSS code it cannot work on server side after building with Rollup. Rollup currently uses `document.createElement('style')` to inject CSS so it fails on the server. Actually I've got no idea what's the best way to handle this.

**Comment by Justineo** [comment]
As I mentioned in Justineo/vue-awesome#43, maybe users shouldn't use packed version if they want SSR.

**Comment by Ornhoj** [comment]
@znck What is the current status for this bug?
